### DISCUSSION TITLE
[TPA] Deploy WebApp (AWS)

### DISCUSSION THREAD

**Discussion by eguiggiaro** [dds-jv/foro#106] [Consultas de herramientas] [answered]
Hola,

¿Cómo están?

Con el equipo estamos probando varias alternativas para deployar la app, y avanzando con AWS (ya pudimos levantar la base en un RDS), nos encontramos con un problema con el servicio Elastic Beanstalk.

- Lo primero que hicimos fue generar un WAR desde mvn (mvn install) y nos creo correctamente el archivo: cuando descomprimimos vemos una estructura que parece estar correcta.

- Lo segundo: crear un entorno Java 8 en Elastic Beanstalk y subir el WAR. **El resultado**: recibimos un error **502 - BAD GATEWAY** (http://gesoc1.us-east-1.elasticbeanstalk.com/) y no logramos dar con la causa.

**Algunas hipótesis que fuimos revisando:**

- El puerto que escucha por defecto AWS en este servicio es el 5000. Hicimos cambios para probar esto, pero la respuesta sigue siendo la misma.
- La estructura del WAR no es correcta. Probamos entonces creando un JAR para ver si levantaba, y no tenemos éxito.

¿Alguna idea de qué puede estar pasando? ¿Tendríamos que modificar la estructura del proyecto Spark para poder deployarlo?

Gracias!

**Comment by gastonprieto** [comment]
Buenas @eguiggiaro !

Nunca use ElasticBeanstalk para Java, pero..

* Si, el War tiene que tener una estructura particular, con un web.xml y otros archivos en otros lugares, no tengo presente la lista, pero buscate como deberia ser la estructura.

* Esta parece ser la mas simple para como tienen ahora la aplicación,  siguieron esta guía? https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/java-se-platform.html

No tienen algunos logs que les deje ver que esta pasando en ese servicio?

-------------

En general cuando les recomendamos AWS es para que usen EC2 como maquina y usen el servicio como un IAAS, entonces uds instalan todo el entorno y corren los procesos como desean

**Comment by eguiggiaro** [comment, marked as answer]
Buenas!

Finalmente, encaramos por el lado de Heroku. Y sí es cierto que con EC2 tal vez era más sencillo que usando EB.

Gracias por la data!!
